cpp/src/arrow/record_batch.cc

```diff
@@ -59,10 +59,11 @@ class SimpleRecordBatch : public RecordBatch {
  public:
   SimpleRecordBatch(std::shared_ptr<Schema> schema, int64_t num_rows,
                     std::vector<std::shared_ptr<Array>> columns,
+                    DeviceAllocationType device_type = DeviceAllocationType::kCPU,
                     std::shared_ptr<Device::SyncEvent> sync_event = nullptr)
       : RecordBatch(std::move(schema), num_rows),
         boxed_columns_(std::move(columns)),
-        device_type_(DeviceAllocationType::kCPU),
+        device_type_(device_type),
         sync_event_(std::move(sync_event)) {
     if (boxed_columns_.size() > 0) {
       device_type_ = boxed_columns_[0]->device_type();
```

Comment:
It's a bit strange to have a default CPU device in the signature, while we then actually get it from the array. Because then it seems that a user can set this, while it will actually be ignored? (except for the case of 0 columns)

What was the reason you added this? Because it needed to be set from Python? (but also there the arrays should already have a device type, so was this only for the case of 0 columns, e.g. after dropping the other columns)

Reply:
Yep, you're right. I'm going to remove it because it is unnecessary.